src/app/auth/services/session-storage.service.ts

```diff
@@ -7,15 +8,18 @@ const TOKEN = 'SESSION_TOKEN'; // Use this constant for the session storage entr
   providedIn: 'root',
 })
 export class SessionStorageService {
+  constructor(@Inject(WINDOW) private window: Window) {}
+
   setToken(token: string) {
-    // Add your code here
+    this.window.sessionStorage.setItem(TOKEN, token);
+    this.getToken();
```

Comment:
Are you sure this getToken needs to be invoked here?

Reply:
it is not, for sure